geopyspark/geotrellis/catalog.py

```diff
@@ -423,19 +423,22 @@ def write(uri,
             be in camel case. If both options and keywords are set, then the options will be used.
     """
 
+    if not tiled_raster_layer.zoom_level:
+        raise ValueError("The given layer does not have a zoom_level", tiled_raster_layer)
```

Comment:
Just so I understand, you're saying that if a `zoom` is not given to `query` then the resulting `TiledRasterLayer` should have a `zoom_level` of `None`, right? I think that's a bit confusing from an API standpoint. If I were doing a read I'd expect the resulting `TiledRasterLayer` to have the `zoom_level` I gave it. In the case that I didn't supply a `zoom`, I'd expect something back with a `zoom_level` of 0. Plus, it's a bit odd that `query("spatial", my_uri, layer_name, layer_zoom=0)` and `query("spatial", my_uri, layer_name)` would return different `TiledRasterLayer`s even though they both read from the same catalog and zoom.